### DISCUSSION TITLE
Include minimum required WordPress version in plugin composer.json

### DISCUSSION THREAD

**Discussion by grappler** [outlandishideas/wpackagist#262] [Ideas]
Tablepress just increased their minimum required WordPress version to `4.9.1`.

The plugin update was not shown in the dashboard but it was downloaded ok via `composer update`.

It would be cool if wpackagist could parse the readme file and add the WordPress dependancy.

In the same pass, it would be possible to get the minimum PHP version too. https://make.wordpress.org/plugins/2017/08/29/minimum-php-version-requirement/

**Comment by khromov** [comment]
@grappler Great idea. The issue we would need to overcome is how to handle the situation where someone does not use the `johnpbloch/wordpress` package as they are either using a different package or they actually don't use Composer for WP core. 

One possible solution could be the "suggested package" syntax. https://getcomposer.org/doc/04-schema.md#suggest

**Comment by grappler** [comment]
Just got bitten by TablePress again where the minimum supported version of WordPress was increased to WP 5.3.

**Comment by szepeviktor** [comment]
me by Yoast SEO

wpackagist could provide `wpackagist/wordpress-core` and simply use Composer's constrains

**Comment by NoelLH** [comment]
Converted to a Discussion for now because I don't think we have a clear, specific way forward yet.

I don't think supplying our own WordPress core for everybody can work, especially as for years we have recommended others in the readme. Surely we don't want to break backwards compatibility and force a new core WP installation method for everyone who has been following that advice?

`suggests` could work fine but you can definitely also miss it in the Composer output – it doesn't enforce anything as far as I know.

**Comment by grappler** [comment]
Would it be possible to activate the feature through a parameter? So people using `johnpbloch/wordpress` could enable a feature that would then provide the dependency requirement. Otherwise wpackagist would function as it does today.

**Reply by NoelLH** [reply]
Yep sounds like it could work as an opt-in feature this way @grappler, so long as we name any `extra` feature flags in a way that makes it clear which WP core installation method(s) they work with. I won't be able to explore this any time soon but happy to look at any PRs to move this forward.
